crates/bevy_sprite/src/mesh2d/color_material.wgsl

```diff
@@ -23,14 +29,40 @@ fn fragment(
     mesh: VertexOutput,
 ) -> @location(0) vec4<f32> {
     var output_color: vec4<f32> = material.color;
+
 #ifdef VERTEX_COLORS
     output_color = output_color * mesh.color;
 #endif
+
     if ((material.flags & COLOR_MATERIAL_FLAGS_TEXTURE_BIT) != 0u) {
         output_color = output_color * textureSample(texture, texture_sampler, mesh.uv);
     }
+
+    output_color = alpha_discard(material, output_color);
+
 #ifdef TONEMAP_IN_SHADER
     output_color = tonemapping::tone_mapping(output_color, view.color_grading);
 #endif
     return output_color;
 }
+
+fn alpha_discard(material: ColorMaterial, output_color: vec4<f32>) -> vec4<f32> {
+    var color = output_color;
+    let alpha_mode = material.flags & COLOR_MATERIAL_FLAGS_ALPHA_MODE_RESERVED_BITS;
+    if alpha_mode == COLOR_MATERIAL_FLAGS_ALPHA_MODE_OPAQUE {
+        // NOTE: If rendering as opaque, alpha should be ignored so set to 1.0
+        color.a = 1.0;
+    }
+    // TODO MAY_DISCARD
```

Comment:
Should either be fixed or have some context, like "add MAY_DISCARD ifdef" (I had to search to figure out what it meant).

Reply:
Oh, right, I completely forgot that wasn't implemented, I'll implement it before it gets merged.